### PR TITLE
Configuration option to override default hostname in stackdriver_agent plugin

### PR DESCRIPTION
**Pull request by dhrupadb**

### CODE REVIEW

File: src/stackdriver_agent.c [hunk 1]
```diff
@@ -206,9 +214,25 @@ static int sagt_init()
     return result;
 }
 
+static int sagt_config(const char *key, const char *value) {
+  if (strcmp(key, "Hostname") == 0) {
+    hostname = (const char *) sstrdup(value);
+    if (hostname == NULL) {
+      WARNING("%s: sagt_config sstrdup failed. Using defaults.",
```
Comment (igorpeshansky) [inline src/stackdriver_agent.c]: I think you actually want to fail here, rather than silently fall back on the global hostname.

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: src/stackdriver_agent.c (hunk 1 above)
Comment (dhrupadb) [inline src/stackdriver_agent.c]: Done. PTAL

File: src/stackdriver_agent.c [hunk 2]
```diff
@@ -30,6 +30,14 @@
 
 static const char this_plugin_name[] = "stackdriver_agent";
 
+static const char *hostname = NULL;
+
+static const char *config_keys[] = {
+  "Hostname"
```
Comment (igorpeshansky) [inline src/stackdriver_agent.c]: Let's add a trailing comma, for consistency.

File: src/stackdriver_agent.c [hunk 3]
```diff
@@ -206,9 +214,26 @@ static int sagt_init()
     return result;
 }
 
+static int sagt_config(const char *key, const char *value) {
+  if (strcmp(key, "Hostname") == 0) {
+    hostname = (const char *) sstrdup(value);
+    if (hostname == NULL) {
+      ERROR("%s: sagt_config sstrdup failed.",
```
Comment (igorpeshansky) [inline src/stackdriver_agent.c]: Does this now fit on one line?

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: src/stackdriver_agent.c (hunk 2 above)
Comment (dhrupadb) [inline src/stackdriver_agent.c]: Done

File: src/stackdriver_agent.c (hunk 3 above)
Comment (dhrupadb) [inline src/stackdriver_agent.c]: Done.

**Review by dhrupadb — COMMENTED** [review]
PTAL

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Comment by igorpeshansky** [comment]
dhrupadb was at Google at the time.

**Pull request merged**